src/font-customization.test.ts

```
@@ -271,4 +277,304 @@ describe('Font customization unit tests', () => {
+      // 12pt body = 24hp, auto-shrink = 20hp = 10pt
+      expect(extractSzVal(block)).toBe(20);
+    });
+
+    it('per-table directive overrides document-level table font', async () => {
+      const markdown = '---\ntable-font-size: 9\n---\n\n<!-- table-font-size: 7 -->\n\n| A |\n|---|\n| 1 |';
+      const result = await convertMdToDocx(markdown);
+      const JSZip = (await import('jszip')).default;
+      const zip = await JSZip.loadAsync(result.docx);
+      const docContent = await zip.file('word/document.xml')!.async('string');
+      // Per-table override: 7pt = 14hp, should appear as w:sz inside w:rPr
+      expect(docContent).toMatch(/<w:rPr>[^]*?<w:sz w:val="14"\/>/);
```

Biome (2.4.4)

[error] The regular expression includes this negated empty character class.

(lint/correctness/noEmptyCharacterClassInRegex)